pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -406,24 +400,24 @@ func buildTapDeviceMaker(tapName string, isMultiqueue bool, virtLauncherPID int)
 		createTapDeviceArgs = append(createTapDeviceArgs, "--multiqueue")
 	}
 	cmd := exec.Command("virt-chroot", createTapDeviceArgs...)
+
 	return &tapDeviceMaker{
-		virtLauncherSELinuxLabel: virtLauncherSELinuxLabel,
-		virtHandlerSELinuxLabel:  virtHandlerSELinuxLabel,
-		tapMakingCmd:             cmd,
-		tapName:                  tapName,
-		isMultiqueue:             isMultiqueue,
+		tapMakingCmd:    cmd,
+		tapName:         tapName,
+		isMultiqueue:    isMultiqueue,
+		virtLauncherPID: virtLauncherPID,
 	}, nil
 }
 
 func (tmc *tapDeviceMaker) makeTapDevice() error {
-	runtime.LockOSThread()
-	defer func() {
-		_ = selinux.SetExecLabel(tmc.virtHandlerSELinuxLabel)
-		runtime.UnlockOSThread()
-	}()
+	if isSELinuxEnabled() {
+		if err := setVirtLauncherSELinuxContext(tmc.virtLauncherPID); err != nil {
+			return err
+		}
 
-	if err := selinux.SetExecLabel(tmc.virtLauncherSELinuxLabel); err != nil {
-		return fmt.Errorf("failed to switch selinux context to %s. Reason: %v", tmc.virtLauncherSELinuxLabel, err)
+		defer func() {
+			_ = resetVirtHandlerSELinuxContext()
```

Comment:
Why do you read the return value if it is not used?

Reply:
To explicitly indicate to the reader / reviewer that I don't care about the error.